### ISSUE TITLE
Inventory

### ISSUE THREAD

**Issue by mattridgway** [smudge202/clean-living#13]
Requests to engine can:
- add item to inventory
- disallow the same instance of an item to be added twice
- return a list of items in the inventory from through perspective

Later:
- limit the size of the inventory (size, weight, count)
- use items in the inventory
- remove item from inventory
- check when adding an item that it was visible/accessible to the actor
- items can perish/be consumed/damaged through use/time
- update perspective so item is no longer visible when picked up (and inverse when dropped)
- ensure its possible to add the item to the inventory (its within perspective and nothing in the way)

**Comment by herecydev** [comment]
Might be worth thinking about consumable/durable/permanent and some sort of transformed items.

Fruit is consumable
Torch is durable
Axe is permanent
Bottle of water transforms into empty bottle when consumed

Something like that

**Comment by mattridgway** [comment]
some items (like food) might need to be perishable too??

**Comment by herecydev** [comment]
Yep, which I guess makes it durable? Maybe chocolate bar is a better example, but you get where I'm going.

**Comment by smudge202** [comment]
However, inventory shouldn't be responsible for any of the transforms listed above.  The items/actors should handle this themselves?  A bottle of water for example would subscribe for a consume event of itself and update accordingly?

**Comment by herecydev** [comment]
I'm not sure if hydration should tell of the transform, If anything that falls under inventory (to me at least)

**Comment by mattridgway** [comment]
updated initial comment to reflect more recent discussions

**Comment by smudge202** [comment]
Given the OP edits, bear in mind we still haven't created any "Perspectives".  We're aware of them being a structure we would like, but not where they belong and how they look.

**Comment by mattridgway** [comment]
Are we still considering that all items in the world should be actors?

**Comment by smudge202** [comment]
They should probably all be something, yes.  _Actors_?  Maybe.  Whatever makes sense.
